adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapter.java

```diff
@@ -147,28 +148,33 @@ public void setWebClient(final WebClient webClient) {
                 return result.future();
             }
 
-            return mapMessageRequest(ctx, targetAddress, message, mapperEndpoint);
+            final VertxHttpHeaders headers = new VertxHttpHeaders();
+            registratonInfo.iterator().forEachRemaining(stringObjectEntry -> headers.add(stringObjectEntry.getKey(),
+                    stringObjectEntry.getValue()));
+
```

Comment:
what is this good for?

Reply:
It allows for some configuration in the defaults to be passed on to the mapper. I think this can be useful if you're writing your own mapping.

Reply:
The *defaults* have well defined semantics and I would rather not encourage people to abuse them for something else. If we need to be able to forward properties that have been registered for the device then these should probably come from the *ext* properties ...

Reply:
Ok. We are currently using those `defaults` to configure our custom mapping after the adapters because the `ext` properties are not passed on.
We are coming back to the discussion on Gitter... Also the reason why I asked you on Gitter. Can we just pass on all `ext` properties from now on?